### PR TITLE
feat: support new and legacy endpoints for URL Shortener MS project, cleanup

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)

Partially addresses https://github.com/freeCodeCamp/freeCodeCamp/issues/41529

Live example with these changes: https://urlshortener-update.scissorsneedfoo.repl.co

### CODE REVIEW

File: url-shortener-microservice/server.js [hunk 1]
```diff
@@ -25,13 +23,16 @@ app.get('/', (req, res) => {
...
 });
 
-app.post('/api/shorturl/new', urlHandler.addUrl);
-  
-app.get('/api/shorturl/:shurl', urlHandler.processShortUrl);
+// support legacy paths
+app.post(['/api/shorturl', '/api/shorturl/new'], urlHandler.addUrl);
 
+app.get(
+  ['/api/shorturl/:shurl', '/api/shorturl/:shurl'],
```
Comment (nhcarrigan) [inline url-shortener-microservice/server.js:30]: What's the difference between these two?

Comment (scissorsneedfoodtoo) [reply url-shortener-microservice/server.js:30]: Oops, that's a mistake. Thanks for catching that.

Originally I was going to shorten this and the POST paths to `/api/...`, but decided against it in the end. With the `shorturl` resource, the GET path won't change, and just the POST path should support the legacy and new paths.

I'll go in and fix this now.

**Review by gikf — APPROVED** [review]
LGTM :+1:

Tested adding new url, redirecting, trying to use wrong route url and wrong short_url.
Tested also against current project tests and locally after updating them with https://github.com/freeCodeCamp/freeCodeCamp/pull/41784.

**Review by nhcarrigan — APPROVED** [review]
Looks good to me - will need to hold on merge until https://github.com/freeCodeCamp/freeCodeCamp/pull/41784 hits production.

**Comment by scissorsneedfoodtoo** [comment]
Awesome, thanks for all the testing you did @gikf and @nhcarrigan.

**Comment by gikf** [comment]
Merged it, as related PR made on to the `prod-current`. Not sure if there's something more needed to do to make changes be shown on the demo project.

**Pull request merged**